pkgs/development/tools/clickable/default.nix

```
@@ -0,0 +1,39 @@
+{ lib
+, fetchFromGitLab
+, buildPythonPackage
+, cookiecutter
+, requests
+, pyyaml
+, jsonschema
+, argcomplete
+}:
+buildPythonPackage rec {
+  pname = "clickable";
+  version = "7.4.0";
+
+  src = fetchFromGitLab {
+    owner = "clickable";
+    repo = "clickable";
+    rev = "v${version}";
+    sha256 = "sha256-QS7vi0gUQbqqRYkZwD2B+zkt6DQ6AamQO7sihD8qWS0=";
+  };
+
+  propagatedBuildInputs = [
+    cookiecutter
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  # Logging test try to write to ~/.clickable/clickable.log
+  doCheck = false;
```

```suggestion
  preCheck = ''
    # Tests use the home directory
    export HOME="$(mktemp -d)"
  ''
```
This should allow the program to access a home directory

---

when I set that, the logging tests fail in another way:

<details>
<summary>Log
</summary>
```
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.9-clickable-7.4.0.drv-0/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 29, in <module>
    setup(
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 148, in setup
    return run_commands(dist)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
    dist.run_commands()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 1214, in run_command
    super().run_command(command)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 224, in run
    self.run_tests()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 227, in run_tests
    test = unittest.main(
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 124, in parseArgs
    self._do_discovery(argv[2:])
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 244, in _do_discovery
    self.createTests(from_discovery=True, Loader=Loader)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 154, in createTests
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 349, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 405, in _find_tests
    tests, should_recurse = self._find_test_path(
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 483, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 57, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 214, in loadTestsFromName
    raise TypeError("don't know how to make test from: %s" % obj)
TypeError: don't know how to make test from: <Logger clickable (DEBUG)>
```
</details>

I am a bit time constrained at the moment, and don't have the ability to dig deeper (: